lib/chai/core/assertions.js

```diff
@@ -1402,7 +1402,8 @@ module.exports = function (chai, _) {
       }
     }
 
-    if (errMsgMatcher) {
+    if (errMsgMatcher !== undefined && errMsgMatcher !== null &&
+        caughtErr !== undefined && caughtErr !== null) {
```

Comment:
Recommend refactor 1405-1406 to:

```js
    if (caughtErr && errMsgMatcher != null) {
```

Reasons:

- `caughtErr` doesn't need empty-string protection; it's either an error or undefined, so a simple truthy test should suffice.
- It's generally considered acceptable to use a loose equality comparison when checking for the `undefined` or `null` combo.